lib/gRpcServer.js

```diff
@@ -40,26 +41,62 @@ class GRpcServer {
   }
 
   run() {
-    const server = new grpc.Server();
-
+    this.server = new grpc.Server();
     this.configureRouter();
 
     this.services = Definition.loadServices(this.packageName);
 
     Object.keys(this.services).forEach((resource) => {
       if (app.controllers[`${resource}Controller`]) {
-        const handlers = getMethodNames(app.controllers[`${resource}Controller`]).reduce((mappings, key) => {
+        const handlers = getMethodNames(
+          app.controllers[`${resource}Controller`]
+        ).reduce((mappings, key) => {
           mappings[key] = this.generateRequestHandler(resource, key);
           return mappings;
         }, {});
 
-        server.addService(this.services[resource].service, handlers);
+        this.server.addService(this.services[resource].service, handlers);
       }
     });
 
-    server.bind(`${this.ip}:${this.port}`, grpc.ServerCredentials.createInsecure());
+    this.server.bind(
+      `${this.ip}:${this.port}`,
+      grpc.ServerCredentials.createInsecure()
+    );
     log('sl-express', 'info', { message: `Starting grpc server on 0.0.0.0:${this.port}` });
-    return server.start();
+    return this.server.start();
+  }
+
+  stop() {
+    if (!this.server) return;
+
+    return new Promise((resolve, reject) => {
+      const timeoutID = setTimeout(() => {
+        log('sl-express', 'error', {
+          message: `Timeout error while trying to stop grpc server`,
+        });
+
+        reject(new Error('Timeout error while trying to stop grpc server'));
+      }, 1000 * 5);
```

Comment:
let's set this to 60 seconds, and maybe make that a ENV

Reply:
ok, I will create a .env file and add a variable maybe called like `SHUTDOWN_GRPC_SERVER_TIMEOUT_PERIOD`?